TrainingExtensions/torch/src/python/aimet_torch/qc_quantize_op.py

```diff
@@ -840,11 +857,14 @@ def forward(self, *inputs):
         # Quantize inputs
         torch_inputs = custom_tensor_utils.to_torch_tensor(inputs)
         quantized_inputs = self._quantize_activation(torch_inputs, self.input_quantizers, 'input')
+        quantized_inputs = custom_tensor_utils.to_custom_tensor(inputs, quantized_inputs)
+
+        forward_fn = _default_forward
+        if torch.is_grad_enabled() and is_recompute_enabled():
+            layer_type = type(self._module_to_wrap)
```

Comment:
Hi @quic-kyunggeu.. There is a potential downside to determining layer type using type() instead of isinstance(). I am approving the PR. We can refactor later.

Reply:
Actually it was a bit intentional because I wanted to apply this only to nn.Linear but not to its subclasses.
We have a good knowledge about nn.Linear and that it doesn't cause any side effect when merged with weight quantizers, but we don't know anything about its potential subclasses.
We can talk about this further later